### PR TITLE
Add  websocket Command  "SetMoveToTarget";

### PR DESCRIPTION
**Pull request by HassenLin**
## Short Description:
Add Set Destination in websocket feature for Visualizer Tool  .

## Fixes (provide links to github issues if you can):
Add PoGo.NecroBot.CLI\WebSocketHandler\SetDestinationHandler.cs
Modify PoGo.NecroBot.Logic\Tasks\UseNearbyPokestopsTask.cs

Online Visualizer Tool : http://hassenlin.byethost12.com/
Source : https://github.com/HassenLin/necrobotvisualizer

### CODE REVIEW

**Comment by samuraitruong** [comment]
Need fix logic for GPX Path . because it will break GPX Path rules.

**Comment by HassenLin** [comment]
Can you point out which segment will break GPX Path rules.
In my design GPX will not enable "set destination", because all position already set in GPX file, it will useless in GPX mode.

File: PoGo.NecroBot.CLI/WebSocketHandler/ActionCommands/ForceUsePokestop.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+        public SetDestination()
+        {
+            Command = "ForceUsePokestop";
+        }
+
+        public async Task Handle(ISession session, WebSocketSession webSocketSession, dynamic message)
+        {
+            Logic.Tasks.UseNearbyPokestopsTask.ForceUsePokestopLat = (double)message.lat;
+            Logic.Tasks.UseNearbyPokestopsTask.ForceUsePokestopLng = (double)message.lng;
```
Comment (samuraitruong) [inline PoGo.NecroBot.CLI/WebSocketHandler/ActionCommands/ForceUsePokestop.cs]: Could you change this to 1 single function to accept all parameter
  Logic.Tasks.UseNearbyPokestopsTask.UpdateCustomTarget(....)

File: PoGo.NecroBot.Logic/Tasks/UseNearbyPokestopsTask.cs [hunk 2]
```diff
@@ -132,6 +145,24 @@ private static bool SearchThresholdExceeds(ISession session)
                     await eggWalker.ApplyDistance(distance, cancellationToken);
                 }
 
+                if (ForceUsePokestopEnabled && ForceUsePokestopAccept)
```
Comment (samuraitruong) [inline PoGo.NecroBot.Logic/Tasks/UseNearbyPokestopsTask.cs]: Not required , however I would ask you to move the logic into a separate task, then only need to call it in usenearbypokestop task
something like below. that will make the code easy to maintain later.
`
 while (pokestopList.Any())
            {
                await MoveToTask.Execute(session);

File: PoGo.NecroBot.CLI/WebSocketHandler/ActionCommands/ForceUsePokestop.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace PoGo.NecroBot.CLI.WebSocketHandler.ActionCommands
+{
+    public class SetDestination : IWebSocketRequestHandler
+    {
+        public string Command { get; private set;}
+
+        public SetDestination()
+        {
+            Command = "ForceUsePokestop";
```
Comment (samuraitruong) [inline PoGo.NecroBot.CLI/WebSocketHandler/ActionCommands/ForceUsePokestop.cs]: I think the command name should be something more readable "SetMoveToTarget" . I think we also get rid of using fake pokestop. we can call session.navigation.move(your set location.)

File: PoGo.NecroBot.CLI/WebSocketHandler/ActionCommands/ForceUsePokestop.cs (hunk 1 above)
Comment (HassenLin) [inline PoGo.NecroBot.CLI/WebSocketHandler/ActionCommands/ForceUsePokestop.cs]: OK

File: PoGo.NecroBot.Logic/Tasks/UseNearbyPokestopsTask.cs (hunk 2 above)
Comment (HassenLin) [inline PoGo.NecroBot.Logic/Tasks/UseNearbyPokestopsTask.cs]: OK

File: PoGo.NecroBot.CLI/WebSocketHandler/ActionCommands/ForceUsePokestop.cs (hunk 3 above)
Comment (HassenLin) [inline PoGo.NecroBot.CLI/WebSocketHandler/ActionCommands/ForceUsePokestop.cs]: OK

Comment (HassenLin) [inline PoGo.NecroBot.CLI/WebSocketHandler/ActionCommands/ForceUsePokestop.cs]: OK

File: PoGo.NecroBot.Logic/Tasks/UseNearbyPokestopsTask.cs [hunk 4]
```diff
@@ -106,13 +102,10 @@ private static bool SearchThresholdExceeds(ISession session)
...
-                    if (ForceUsePokestopEnabled)
-                    {
-                        ForceUsePokestopAccept = true;
-                        fortInfo.Name = "User Destination.";
-                        fortInfo.Latitude = pokeStop.Latitude = ForceUsePokestopLat;
-                        fortInfo.Longitude = pokeStop.Longitude = ForceUsePokestopLng;
-                    }
+
+                    // Will modify Lat,Lng and Name to fake position
+                    SetMoveToTargetTask.checkSetMoveToTargetStatus(ref fortInfo, ref pokeStop);
```
Comment (samuraitruong) [inline PoGo.NecroBot.Logic/Tasks/UseNearbyPokestopsTask.cs]: checkSetMoveToTargetStatus should be =>CheckSetMoveToTargetStatus

File: PoGo.NecroBot.Logic/Tasks/UseNearbyPokestopsTask.cs [hunk 5]
```diff
@@ -128,8 +121,8 @@ private static bool SearchThresholdExceeds(ISession session)
                         LocationUtils.getElevation(session, pokeStop.Latitude, pokeStop.Longitude)),
                     async () =>
                     {
-                        if (ForceUsePokestopEnabled && !ForceUsePokestopAccept)
-                            return true;
+                        if (SetMoveToTargetTask.isStopforSetMoveToTarget())
```
Comment (samuraitruong) [inline PoGo.NecroBot.Logic/Tasks/UseNearbyPokestopsTask.cs]: Fix naming convention

Comment (HassenLin) [inline PoGo.NecroBot.Logic/Tasks/UseNearbyPokestopsTask.cs]: OK, Thank you.

**Pull request merged**